### ISSUE TITLE
[Linux Ubuntu 22.04] Builds newer than 2024-09-26 fail to run (GLIBC_2.38 and GLIBC_2.39 not available)

### ISSUE THREAD

**Issue by heckert-gamedev** [ruffle-rs/ruffle#18341]
### Describe the bug

I found that the CI changes from 2024-09-29 (upgrading to 24.04 build images) broke the Linux version on my machine running Ubuntu 22.04:

```
~ /Downloads/Ruffles/20240929$ ldd ruffle 
./ruffle: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found (required by ./ruffle)
./ruffle: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.39' not found (required by ./ruffle)
        linux-vdso.so.1 (0x00007ffe1c137000)
        libasound.so.2 => /lib/x86_64-linux-gnu/libasound.so.2 (0x0000767f5cefd000)
        libudev.so.1 => /lib/x86_64-linux-gnu/libudev.so.1 (0x0000767f5f06d000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x0000767f5f04d000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x0000767f5ce16000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x0000767f5ca00000)
        /lib64/ld-linux-x86-64.so.2 (0x0000767f5f0b7000)
```

The latest version of libc6 for Ubuntu 22.04 (Jammy) is 2.35 and there are no backports either.

The Ruffle binary from the previous build (2204-09-26) runs fine.

### Expected behavior

Runtime dependencies are satisfied and Ruffle can be started.

### Content Location

This issue is independent of a specific flash file.

### Affected platform

Desktop app

### Operating system

Ubuntu 22.04 (Jammy Jellyfish)

**Comment by kjarosh** [comment]
Unfortunately, compatibility with glibc is a long-stading and known issue. It's very unlikely we'll be able to provide binaries compatible with every version of every distribution.

We can recommend using the Flatpak version of Ruffle which guarantees compatibility across different distributions, even those with older versions of glibc: https://flathub.org/apps/rs.ruffle.Ruffle Ruffle will also update automatically when installed through Flatpak.

**Comment by heckert-gamedev** [comment]
Right... I had to install the `flatpak` utility, then define the `flathub` repository, then install another 2GB of stuff along a nice little 40MB application. And then I had to hunt down where the flatpaked version had its data directory to copy the save files over.

Or I can free up that disk space and keep stuck with the September build (which is about half a year more recent than the one I used before and has quite a few noticeable improvements...)

**Comment by sombraguerrero** [comment]
As a courtesy, I do regularly build from source for older Ubuntu versions and publicly post it. I've been thinking of finally moving on from 20.04 and 22,04 would be the next logical progression of that.

**Comment by StevanWhite** [comment]
It is a weakness of a software product to be so sensitive to library versions that it will run only with two minor versions, and no others.  It is very suspicious.

The problem is more widespread that has been mentioned in this report.  There are current distros for which glibc is too *new* as well as too *old* for the executable to run.

The current builds do not run on the most recent Linux Mint 

    GNU C Library (Ubuntu GLIBC 2.35-0ubuntu3.8) stable release version 2.35.

or Manjaro linux, for the opposite reason to the one reported:

    GNU C Library (GNU libc) stable release version 2.40

**Comment by sombraguerrero** [comment]
@StevanWhite We are very aware that this is a less than ideal shortcoming and equally aware that it is a wider spread issue than one ticket describes, but there is nothing nefarious behind this. We have simply built a product with a lot of dependencies, common enough that the *nix world has mechanisms to compensate for the problem, as you doubtless know. We do have a Flatpak available and building from source is always an option.

**Comment by StevanWhite** [comment]
@sombraguerrero, you wrote "...but there is nothing nefarious behind this"

Sorry, I didn't mean to be cryptic.  When I said this kind of thing was suspicious,
I meant, it sounds like programming around a bug in a library.

**Comment by kjarosh** [comment]
The issue is not static linking, it's the opposite: dynamic linking. The reason why glibc is a shared library is because of its LGPL license and improved usage of resources.
